src/mo_frontend/effect.ml

```diff
@@ -25,13 +25,32 @@ let typ phrase = phrase.note.note_typ
 let eff phrase = phrase.note.note_eff
 
 let is_triv phrase  =
-    eff phrase = T.Triv
+  eff phrase = T.Triv
+
+let is_shared_func exp =
+  T.(match typ exp with
+    | Func (Shared _, _, _, _, _) -> true
+    | _ -> false)
+
+let is_local_async_func exp =
+ T.(match typ exp with
+   | Func (Local, Returns,
+      { sort = Scope; _ }::_,
+       _,
+      [Async (Fut, Var (_ ,0), _)]) -> true
+   | _ -> false)
+
+let is_async_call exp1 inst exp2 =
```

Comment:
`inst` and  `exp2` are irrelevant for the result

Reply:
Yes, but its a predicate on calls, so I've include them.